### PR TITLE
Aaronmgdr/invite page

### PR DESCRIPTION
**Pull request by aaronmgdr**
opening this draft PR as changes are happening to the faucet and id like them to be checked sooner rather than later 

### Description
GOAL
Add ability to choose which OS your phone is on the websites Invite-Faucet Page. (celo.org/build/wallet)

Method
- add a picker to web UI
- pass a `mobileOS` value to firebase and select the right link to send in message based on this. 

note
it looked like we already duplicated some types/interfaces between faucet and web projects so i continued that

### Tested
UI wise yes, not for the functionality yet

### Other changes

deduplicate code for sending dollars in the faucet project 

### Related issues

- Fixes https://github.com/celo-org/celo-labs/issues/144   https://github.com/celo-org/celo-labs/issues/151

### Backwards compatibility

if no OS is provided it should just still send the android version of link

### CODE REVIEW

**Comment by aaronmgdr** [comment]
@trianglesphere @jmrossy @mcortesi  would appreciate a look at the faucet changes

File: packages/faucet/src/database-helper.ts [hunk 1]
```diff
@@ -26,11 +26,19 @@ export enum RequestType {
   Invite = 'Invite',
 }
 
+enum MobileOS {
+  'andriod' = 'android',
```
Comment (mcortesi) [inline packages/faucet/src/database-helper.ts]: typo `andriod`. Also, why the `''` in the key name?

File: packages/faucet/src/database-helper.ts [hunk 2]
```diff
@@ -26,11 +26,19 @@ export enum RequestType {
...
+  ios = 'ios',
+}
+
 export interface RequestRecord {
   beneficiary: Address
   status: RequestStatus
-  txHash?: string
```
Comment (mcortesi) [inline packages/faucet/src/database-helper.ts:32]: should we remove this from database-rules too?

File: packages/faucet/src/database-helper.ts [hunk 3]
```diff
@@ -141,15 +141,43 @@ function buildHandleInvite(request: RequestRecord, snap: DataSnapshot, config: N
...
+}
+
+function messageText(inviteCode: string, request: RequestRecord) {
+  return `Hello! Thank you for joining the Celo network. Your invite code is: ${inviteCode} Download the app at ${downloadLink(
+    request.mobileOS as MobileOS
+  )}`
+}
+
+const IOS_URL = ''
+const ANDRIOD_URL = 'https://play.google.com/store/apps/details?id=org.celo.mobile.alfajores'
```
Comment (mcortesi) [inline packages/faucet/src/database-helper.ts]: typo andriod

File: packages/faucet/src/database-helper.ts [hunk 4]
```diff
@@ -141,15 +141,43 @@ function buildHandleInvite(request: RequestRecord, snap: DataSnapshot, config: N
...
+  return dollarTxHash
+}
+
+function messageText(inviteCode: string, request: RequestRecord) {
+  return `Hello! Thank you for joining the Celo network. Your invite code is: ${inviteCode} Download the app at ${downloadLink(
+    request.mobileOS as MobileOS
+  )}`
+}
+
+const IOS_URL = ''
```
Comment (mcortesi) [inline packages/faucet/src/database-helper.ts]: missing url

File: packages/faucet/src/database-helper.ts [hunk 5]
```diff
@@ -141,15 +141,43 @@ function buildHandleInvite(request: RequestRecord, snap: DataSnapshot, config: N
...
+const ANDRIOD_URL = 'https://play.google.com/store/apps/details?id=org.celo.mobile.alfajores'
+
+function downloadLink(mobileOS: MobileOS) {
+  if (mobileOS === MobileOS.ios) {
```
Comment (mcortesi) [inline packages/faucet/src/database-helper.ts]: nit. Use `x == y ? z : k`

File: packages/faucet/src/database-helper.ts [hunk 6]
```diff
@@ -116,17 +120,13 @@ function buildHandleInvite(request: RequestRecord, snap: DataSnapshot, config: N
...
-    console.info(`req(${snap.key}): Gold Transaction Sent. txhash:${goldTxHash}`)
-    await snap.ref.update({ goldTxHash })
-    await goldTx.waitReceipt()
-
-    const dollarTx = await celo.transferDollars(tempAddress, config.inviteDollarAmount)
-    const dollarTxHash = await dollarTx.getHash()
-    console.info(`req(${snap.key}): Dollar Transaction Sent. txhash:${dollarTxHash}`)
-    await snap.ref.update({ dollarTxHash })
-    await dollarTx.waitReceipt()
+    // Dont need gold as fees are paid in dollars
```
Comment (mcortesi) [inline packages/faucet/src/database-helper.ts]: for this to work you need to specify the `GasCurrency` when sending the TX. So probably this is not working

**Review by mcortesi — COMMENTED** [review]
Reviewed only `faucet` changes. Left a few comments.

File: packages/faucet/src/database-helper.ts (hunk 3 above)
Comment (aaronmgdr) [reply packages/faucet/src/database-helper.ts]: wow. thanks

File: packages/faucet/src/database-helper.ts (hunk 6 above)
Comment (aaronmgdr) [reply packages/faucet/src/database-helper.ts]: ah ok

**Comment by aaronmgdr** [comment]
thanks for the review @mcortesi

File: packages/faucet/src/database-helper.ts (hunk 1 above)
Comment (aaronmgdr) [reply packages/faucet/src/database-helper.ts]: no good reason just moving fast

File: packages/faucet/src/database-helper.ts (hunk 2 above)
Comment (aaronmgdr) [reply packages/faucet/src/database-helper.ts:32]: if i'm understanding this right, yes and we should add `goldTxHash` `dollarTxHash` and `escrowTxHash`

those three are already used on the `ref`

Comment (mcortesi) [reply packages/faucet/src/database-helper.ts:32]: yes

**Review by cmcewen — APPROVED** [review]

**Pull request merged**